### PR TITLE
fix NPE in PolygonShape.set when no Vec2ArrayPool is given

### PR DESCRIPTION
**Pull request by jbellis**
Note that the cast on line 133 is necessary for Kotlin to figure out that the actual fix on line 147 is valid

### CODE REVIEW

File: kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt [hunk 1]
```diff
@@ -212,9 +213,6 @@ class PolygonShape : Shape(ShapeType.POLYGON) {
 
         // Copy vertices.
         for (i in 0 until count) {
-            if (vertices[i] == null) {
-                vertices[i] = Vec2()
-            }
```
Comment (soywiz) [inline kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt:217]: Why is this removed?

File: kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt [hunk 2]
```diff
@@ -128,10 +128,11 @@ class PolygonShape : Shape(ShapeType.POLYGON) {
...
 
         // Perform welding and copy vertices into local buffer.
-        val ps = if (vecPool != null)
-            vecPool[Settings.maxPolygonVertices]
+        val ps : Array<Vec2?>
+        if (vecPool != null)
+            ps = vecPool[Settings.maxPolygonVertices] as Array<Vec2?>
         else
-            arrayOfNulls<Vec2>(Settings.maxPolygonVertices)
+            ps = arrayOfNulls<Vec2>(Settings.maxPolygonVertices)
```
Comment (soywiz) [inline kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt:135]: This change shouldn't be required, right?

**Comment by soywiz** [comment]
Can you also add your failing test case here?
https://github.com/korlibs/korge-next/tree/f01a13a932e4ed4f96c39a20604df902ebf7ef38/kbox2d/src/commonTest/kotlin/org/jbox2d/utests

Something like:

```kotlin
class PolygonShapeTest {
  @Test
  fun testPolygonShapeTest() {
    val shape = PolygonShape()
    var zero = Vec2(0f, 0f)
    shape.set(arrayOf(zero, zero, zero), 3) // This shouldn't throw
    assertEquals(...)
  }
}
```

**Comment by soywiz** [comment]
Fixes https://github.com/korlibs/korge/issues/445

File: kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt (hunk 2 above)
Comment (jbellis) [reply kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt:135]: Above: 

Note that the cast on line 133 is necessary for Kotlin to figure out that the actual fix on line 147 is valid

**Comment by jbellis** [comment]
test added to PR

File: kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt (hunk 1 above)
Comment (soywiz) [reply kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt:217]: What makes you think it is unnecessary in all the cases?

File: kbox2d/src/commonTest/kotlin/org/jbox2d/utests/PolygonShapeTest.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+
+class PolygonShapeTest {
+    @Test
+    fun test() {
+        val shape = PolygonShape()
+        val vertexArray = arrayOf(Vec2(1f, 1f), Vec2(1f, 0f), Vec2(0f, 0f))
+        shape.set(vertexArray, vertexArray.size)
+
+        assertEquals(shape.count, vertexArray.size)
```
Comment (soywiz) [inline kbox2d/src/commonTest/kotlin/org/jbox2d/utests/PolygonShapeTest.kt:18]: Cool. Thanks!

File: kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt (hunk 2 above)
Comment (soywiz) [reply kbox2d/src/commonMain/kotlin/org/jbox2d/collision/shapes/PolygonShape.kt:135]: Now the declaration and the assignment is split but it looks the same to me and it was compiling before. What happens if you restore the original lines there?

**Comment by soywiz** [comment]
Just checked that vectors: Array<Vec2> is already initialized, so indeed not needed. Going to merge it now.
Thanks!
